InfrastructureManager/agents/azure_agent.py

```diff
@@ -0,0 +1,454 @@
+from azure.common.credentials import ServicePrincipalCredentials
+from azure.mgmt.compute import ComputeManagementClient
+from azure.mgmt.compute.models import CachingTypes
+from azure.mgmt.compute.models import DiskCreateOptionTypes
+from azure.mgmt.compute.models import HardwareProfile
+from azure.mgmt.compute.models import LinuxConfiguration
+from azure.mgmt.compute.models import NetworkProfile
+from azure.mgmt.compute.models import NetworkInterfaceReference
+from azure.mgmt.compute.models import OperatingSystemTypes
+from azure.mgmt.compute.models import OSDisk
+from azure.mgmt.compute.models import OSProfile
+from azure.mgmt.compute.models import SshConfiguration
+from azure.mgmt.compute.models import SshPublicKey
+from azure.mgmt.compute.models import StorageProfile
+from azure.mgmt.compute.models import VirtualHardDisk
+from azure.mgmt.compute.models import VirtualMachine
+from azure.mgmt.compute.models import VirtualMachineSizeTypes
+
+from azure.mgmt.network import NetworkManagementClient
+from azure.mgmt.network.models import AddressSpace
+from azure.mgmt.network.models import IPAllocationMethod
+from azure.mgmt.network.models import NetworkInterfaceIPConfiguration
+from azure.mgmt.network.models import NetworkInterface
+from azure.mgmt.network.models import PublicIPAddress
+from azure.mgmt.network.models import Subnet
+from azure.mgmt.network.models import VirtualNetwork
+
+from haikunator import Haikunator
+
+# AppScale-specific imports
+from agents.base_agent import AgentConfigurationException
+from agents.base_agent import AgentRuntimeException
+from agents.base_agent import BaseAgent
+from utils import utils
+
+class AzureAgent(BaseAgent):
+  """ AzureAgent defines a specialized BaseAgent that allows for interaction
+  with Microsoft Azure. It authenticates using the ADAL (Active Directory
+  Authentication Library).
+  """
+  # The Azure URL endpoint that receives all the authentication requests.
+  AZURE_AUTH_ENDPOINT = 'https://login.microsoftonline.com/'
+
+  # The Azure Resource URL to get the authentication token using client credentials.
+  AZURE_RESOURCE_URL = 'https://management.core.windows.net/'
+
+  # The default Storage Account name to use for Azure.
+  DEFAULT_STORAGE_ACCT = 'appscalestorage'
+
+  # The default resource group name to use for Azure.
+  DEFAULT_RESOURCE_GROUP = 'appscalegroup'
+
+  # The following constants are string literals that can be used by callers to
+  # index into the parameters that the user passes in, as opposed to having to
+  # type out the strings each time we need them.
+  PARAM_APP_ID = 'app_id'
+  PARAM_APP_SECRET = 'app_secret_key'
+  PARAM_CREDENTIALS = 'credentials'
+  PARAM_EXISTING_RG = 'does_exist'
+  PARAM_GROUP = 'group'
+  PARAM_INSTANCE_IDS = 'instance_ids'
+  PARAM_KEYNAME = 'keyname'
+  PARAM_IMAGE_ID = 'image_id'
+  PARAM_REGION = 'region'
+  PARAM_RESOURCE_GROUP = 'resource_group'
+  PARAM_STORAGE_ACCOUNT = 'storage_account'
+  PARAM_SUBCR_ID = 'subscription_id'
```

Comment:
I'd change that to PARAM_SUBSCRIPTION_ID to match the rest of them.
If that makes other usages too long, change to PARAM_SUB**S**CR_ID.